### PR TITLE
Update error message and underscore check

### PR DESCRIPTION
**Pull request by jasmineHundal**
Update error message for `program_set_mismatch?` and add `!` for underscore check as the branch check is an `unless` statement.

### CODE REVIEW

**Comment by jasmineHundal** [comment]
I did think of just doing:
`unless branch_contains_program_set? && branch_contains_prefix? && !branch_name_contains_underscore?`
but `unless` statement with `&&` and `! `gets pretty confusing, this way its easier to understand in my opinion.

File: incent-config/Dangerfile [hunk 1]
```diff
@@ -133,8 +133,8 @@ def branch_contains_prefix?
   prefixes.any? { |prefix| branch_name.start_with?(prefix) }
 end
 
-def branch_name_contains_underscore?
-  @branch_name_contains_underscore ||= branch_name.include?('_')
+def branch_name_contains_no_underscore?
+  @branch_name_contains_underscore ||= !branch_name.include?('_')
```
Comment (scookdev) [inline incent-config/Dangerfile:137]: This should be `@branch_name_contains_no_underscore`

**Review by scookdev — CHANGES_REQUESTED** [review]
Looks great - just need to change the memoized variable name

File: incent-config/Dangerfile (hunk 1 above)
Comment (jasmineHundal) [reply incent-config/Dangerfile:137]: good catch!

**Review by scookdev — APPROVED** [review]

**Pull request merged**